### PR TITLE
Fix NullPointerException when running Kotlin scripts without view

### PR DESCRIPTION
**Pull request by wapmon**
Adding null checks for view for Kotlin script engine to allow running in headless mode again.

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

——

It would be good to add a note to the unreleased section of the CHANGELOG.md as well.

File: zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java [hunk 1]
```diff
@@ -1492,14 +1494,20 @@ private void preInvokeScript(ScriptWrapper script) throws ScriptException {
...
+        ViewDelegate view = getView();
 
+        se.getContext().setWriter(writer);
         // Set the script name as a context attribute - this is used for script level variables
         se.getContext().setAttribute(SCRIPT_NAME_ATT, script.getName(), ScriptContext.ENGINE_SCOPE);
 
         se.put("control", Control.getSingleton());
-        se.put("model", getModel());
-        se.put("view", getView());
+        if (model != null) {
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java:1505]: This should not be needed.

File: zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java [hunk 2]
```diff
@@ -1492,14 +1494,20 @@ private void preInvokeScript(ScriptWrapper script) throws ScriptException {
     private Invocable invokeScriptImpl(ScriptWrapper script) {
         ScriptEngine se = script.getEngine().getEngine();
         Writer writer = getWriters(script);
-        se.getContext().setWriter(writer);
+        Model model = getModel();
+        ViewDelegate view = getView();
 
+        se.getContext().setWriter(writer);
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java:1500]: Revert this.

File: zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java [hunk 3]
```diff
@@ -1492,14 +1494,20 @@ private void preInvokeScript(ScriptWrapper script) throws ScriptException {
...
         // Set the script name as a context attribute - this is used for script level variables
         se.getContext().setAttribute(SCRIPT_NAME_ATT, script.getName(), ScriptContext.ENGINE_SCOPE);
 
         se.put("control", Control.getSingleton());
-        se.put("model", getModel());
-        se.put("view", getView());
+        if (model != null) {
+            se.put("model", model);
+        }
+        if (view != null) {
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java:1508]: `hasView()`

File: zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java [hunk 4]
```diff
@@ -1492,14 +1494,20 @@ private void preInvokeScript(ScriptWrapper script) throws ScriptException {
...
+            se.put("view", view);
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/script/ExtensionScript.java:1509]: `getView()`

**Review by psiinon — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**